### PR TITLE
fix(forms): fix callback API, add groupName prop

### PR DESCRIPTION
**Pull request by eckdanny-osi**
📘 [`CL-261`](https://osi-cwds.atlassian.net/browse/CL-261)

* [x] singular callback interface for `onBlur`/`onChange` on `FormControl`
* [x] 📓update `ButtonGroup` Article (➰)
* [x] 📓 update `Kitchen Sink` Article
* [x] Demonstrates usage w/ `formik` for state mgmt
* [ ] ~❓ validation~ (save for later)
* [ ] ❓ `groupName` (remove?)
* [x] consistent click on `label` focuses the `Control`
* [x] ...
* [x] 🐛 impl `Input#onChange(event, *newValue*)`
* [x] clean up kitchen sink Article copy
* [x] remove the `groupName` stuff
* [x] revert Mask/Parser changes
* [ ] ~Make sure no examples broke (we don't have GUI test framework for this yet)~
* [x] include typings for `reactstrap` components in `@cwds/reactstrap`, remove ambient defs in `libs/forms`
* [x] add build step for copying d.ts files in JS pkgs

### CODE REVIEW

File: apps/www/src/modules/components/articles/Forms/FormsKitchenSinkArticle.mdx [hunk 1]
```diff
@@ -1 +1,13 @@
-TODO
+import KitchenSink from './examples/FormsKitchenSink'
+import KitchenSinkSrc from '!raw-loader!./examples/FormsKitchenSink'
+import { CodeBlock } from '@cwds/docs'
+
+## Synopsis
+
+asdlfkj
```
Comment (mlipardo) [inline apps/www/src/modules/components/articles/Forms/FormsKitchenSinkArticle.mdx]: Can we remove this?
```suggestion
```

File: libs/forms/src/Parsers.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+interface IParser {
```
Comment (mlipardo) [inline libs/forms/src/Parsers.ts]: Same `Interface IParser` at `libs/forms/src/Mask.ts`

**Comment by mlipardo** [comment]
`InputMask.test.js` its test have been skipped.

File: apps/www/src/modules/components/articles/Forms/InputMaskArticle.mdx [hunk 3]
```diff
@@ -3,13 +3,29 @@ import { InputMask, MASKS } from '@cwds/forms'
 import { DemoCard, ReferenceCard, PropTable } from '@cwds/docs'
 import { Alert, Card, CardBody, Row, Col } from '@cwds/components'
 import Docgen from '../ReactDogenInfo'
+const myMask = [
```
Comment (eckdanny-osi) [inline apps/www/src/modules/components/articles/Forms/InputMaskArticle.mdx]: no

File: apps/www/src/modules/components/articles/Forms/InputMaskArticle.mdx [hunk 4]
```diff
@@ -3,13 +3,29 @@ import { InputMask, MASKS } from '@cwds/forms'
...
+  /\d/,
+]
 
 ## Synopsis
 
 Use an `Input Mask` in to capture phone numbers, dates, social security input in forms.
 
 <DemoCard>
-  <InputMask mask={MASKS.PHONE} showMask />
+  <InputMask mask={myMask} showMask />
```
Comment (eckdanny-osi) [inline apps/www/src/modules/components/articles/Forms/InputMaskArticle.mdx]: no

File: apps/www/src/modules/components/articles/Forms/examples/RadioGroupExample.jsx [hunk 5]
```diff
@@ -28,6 +28,7 @@ const RadioGroupExample = ({
               <FormGroup>
                 <Label>{label}</Label>
                 <RadioGroup
+                  groupName="My RadioGroup"
```
Comment (eckdanny-osi) [inline apps/www/src/modules/components/articles/Forms/examples/RadioGroupExample.jsx]: remove the `groupName` stuff plz

File: apps/www/src/modules/components/articles/Forms/examples/RadioGroupValidationExample.jsx [hunk 6]
```diff
@@ -31,6 +31,7 @@ const BasicCheckboxBankExample = ({
             <FormField
               required={required}
               name={name}
+              groupName="My Metastatic Variables"
```
Comment (eckdanny-osi) [inline apps/www/src/modules/components/articles/Forms/examples/RadioGroupValidationExample.jsx]: ditto

File: libs/forms/src/Masks.ts [hunk 7]
```diff
@@ -1,31 +1,41 @@
-export const DATE = [/\d/, /\d/, '/', /\d/, /\d/, '/', /\d/, /\d/, /\d/, /\d/]
+export interface IParser {
```
Comment (eckdanny-osi) [inline libs/forms/src/Masks.ts]: really need to make changes in this file??

File: libs/forms/src/Select/Select.tsx [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+      },
+    }),
+    onChange: () => {},
+    parseValue: reactSelectValue => {
+      if (!reactSelectValue) return ''
+      return Array.isArray(reactSelectValue)
+        ? reactSelectValue.map(({ value }) => value)
+        : reactSelectValue.value
+    },
+    mapValue: (valueOrValues, options) => {
```
Comment (eckdanny-osi) [inline libs/forms/src/Select/Select.tsx]: one might guess the default for mapValues is just identity

**Comment by eckdanny-osi** [comment]
Considering either 1) rebasing to clean up commit history to identify the individual JIRA tickets closed by this PR, or 2) slicing up into smaller PRs

Discussed w/ @karthik-sista at EOD

**Comment by eckdanny-osi** [comment]
closing in favor of #619

**Pull request closed without merging**